### PR TITLE
Make sure upgrade to conan2

### PR DESCRIPTION
**Pull request by shun2wang**
My conan was upgrade to 2.0 version but not really work so let's make sure it. the benefit of version 2.0 is that we can greatly reduce build configuration time on local builds without configuring dependencies again.

Using `pip install --upgrade conan` to upgrade local conan version to 2.0.x, and then using `conan version` to check install status.
maybe not wise but have to bump some pkg version to resolve conflict on conan center.

- Windows:

Visual Studio on my local is VS2022(i.e. VS17 193）in docs:
```
compiler.version=193
tools.microsoft.msbuild:vs_version = 17
```

- Linux:

N/A

- macOS:

I'm not on mac so cannot check it, anyone on mac can help.

### CODE REVIEW

File: Desktop/modules/ribbonbutton.cpp [hunk 1]
```diff
@@ -216,8 +216,8 @@ void RibbonButton::setEnabled(bool enabled)
 	{
 		if(!isSpecial())
 		{
-			if(enabled)	DynamicModules::dynMods()->loadModule(name());
-			else		DynamicModules::dynMods()->unloadModule(name());
+			if(enabled)	DynamicModules::dynMods()->loadModule(_module->name());
```
Comment (JorisGoosen) [inline Desktop/modules/ribbonbutton.cpp]: Did you add this to fix something?

Comment (JorisGoosen) [reply Desktop/modules/ribbonbutton.cpp]: or wait, is this from some other commit then yours?
Maybe you have to rebase it after I just push-forced development

Comment (shun2wang) [reply Desktop/modules/ribbonbutton.cpp]: yes this is not on my PR, let me rebase it.

**Comment by shun2wang** [comment]
@JorisGoosen Are there still problems with Conan2 on mac now? It seems that your second commit is no longer related to Conan😜

**Comment by JorisGoosen** [comment]
Yeah well, I moved the signing identity to its own commit and already made a pr https://github.com/jasp-stats/jasp-desktop/pull/5277

But on macos there are definitely still problems... there is some issue where readstat seemingly links to a version of libiconv, but not the one being linked to JASP...

**Comment by JorisGoosen** [comment]
ill work on it some more this week

**Comment by JorisGoosen** [comment]
https://stackoverflow.com/questions/57734434/libiconv-or-iconv-undefined-symbol-on-mac-osx looks like someone already did the hard work for me figuring what the hell is going on here :p

**Comment by JorisGoosen** [comment]
I still need to test whether the brotli workaround I commented out can be left out or not but beyond that I think it now works on macos

**Comment by JorisGoosen** [comment]
Well, I can build a dmg and run that so I guess the brotli workaround is no longer necessary.
If @Kucharssim @boutinb or @RensDofferhoff could give this a test and verify its not just my mac it works on then we can merge this and move over to the new conan ^^

**Comment by RensDofferhoff** [comment]
RCPP and RInside compilation still fail for me when using a non local signkey.
When using a local sign key it gets through cmake config but fails in compilation.
(I send you the error log)

**Comment by RensDofferhoff** [comment]
Also if we move to conan2 now you will need to update it on the buildbot. More likely than not it will no longer be able to build stable. 
I left bruno  some instructions on how to alter the buildbot set environment so you can set conan2 in PATH.
Take care to test the multibridge mpfr stuff on mac as updating conan version might cause brew to auto update gmp,etc

**Comment by JorisGoosen** [comment]
> Also if we move to conan2 now you will need to update it on the buildbot. More likely than not it will no longer be able to build stable. I left bruno some instructions on how to alter the buildbot set environment so you can set conan2 in PATH. Take care to test the multibridge mpfr stuff on mac as updating conan version might cause brew to auto update gmp,etc

Do we run it with a buildscript?
We could change the path on `stable`

**Comment by JorisGoosen** [comment]
> RCPP and RInside compilation still fail for me when using a non local signkey. When using a local sign key it gets through cmake config but fails in compilation. (I send you the error log)

So yeah the problem here is `clang: warning: no such sysroot directory: 'macosx' [-Wmissing-sysroot]` 
Basically I solved it by setting `SDKROOT` to the output of `xcrun --sdk macosx --show-sdk-path`
This replaces the `macosx` thing with a usable path.

But I dont understand why this is a problem all of a sudden, but I guess conan1 was setting more stuff in the environment than we knew?
Ill see about getting this autodetected though

**Comment by JorisGoosen** [comment]
You could also set 
afbeelding

**Comment by RensDofferhoff** [comment]
Still not working for me maybe @boutinb  has more luck?

**Comment by boutinb** [comment]
When I use conan2 and this PR, I still get an error:
```
Installing the 'RInside' and 'Rcpp' - unsuccessful.
Cmake Error at Tools/CMake/R.cmake:520 (message): 'RInside' installation has failed.
```
If I rerun on a terminal the last command, that is _scripts/install-RInside.R, I get:
```
Frameworks/R.framework/Versions/4.3-x86_64/Resources/bin/R: line 249: /etc/ldpaths: No such file or directory
```
But after some research, this is not the cause of the problem:  R_HOME and R_ARCH are not set, but if I have a look at the R script, I see that it sets R_HOME via R_HOME_DIR, which is self set by JASP_R_HOME, and in the log I can see that R_HOME is set to the right value (R_ARCH is set to nothing), but it can find then the right ldpaths.
But if I rerun now the  _scripts/install-RInside.R script, it just stopped with the message 
```
Killed: 9
```
I remember that this kind of errors occurs with a signing problem....Maybe it does not take the default signing key anymore?
I have to leave, tomorrow better

**Comment by shun2wang** [comment]
> But after some research, this is not the cause of the problem: R_HOME and R_ARCH are not set

Yes, `R_HOME` on Linux is also not be set automatically. we should change R.cmake. on Linux we can find and set default to `usr/lib/R` but I dont know what to set on mac.

**Comment by JorisGoosen** [comment]
> When I use conan2 and this PR, I still get an error:
> 
> ```
> Installing the 'RInside' and 'Rcpp' - unsuccessful.
> Cmake Error at Tools/CMake/R.cmake:520 (message): 'RInside' installation has failed.
> ```
> 
> If I rerun on a terminal the last command, that is _scripts/install-RInside.R, I get:
> 
> ```
> Frameworks/R.framework/Versions/4.3-x86_64/Resources/bin/R: line 249: /etc/ldpaths: No such file or directory
> ```
> 
> But after some research, this is not the cause of the problem: R_HOME and R_ARCH are not set, but if I have a look at the R script, I see that it sets R_HOME via R_HOME_DIR, which is self set by JASP_R_HOME, and in the log I can see that R_HOME is set to the right value (R_ARCH is set to nothing), but it can find then the right ldpaths. 


Yeah I forgot when we ran it in the terminal that you need to set JASP_R_HOME, which ofc is done in the normal build.

**Comment by boutinb** [comment]
Installing the RInside and Rcpp is still unsuccessfull on my Mac Intel.
And from the command line, the _scripts/install-RInside.R script still returns killed: 9.
If I have a look at the messages, I see that it uses well the default signing key...

**Comment by JorisGoosen** [comment]
That is dissapointing...

What do you get if you run the failed RInside command yourself in the terminal?

**Comment by JorisGoosen** [comment]
:'(

Its now finally working on macOs for me, so that is great!

But... 
Windows fails for me...
Ive been searching for the cause but am unsure, maybe it helps to compare conan settings?

My conan profile is:
```
[settings]
arch=x86_64
build_type=Release
compiler=msvc
compiler.cppstd=14
compiler.runtime=dynamic
compiler.version=193
os=Windows
[conf]
tools.microsoft.msbuild:vs_version=17
tools.cmake.cmaketoolchain:generator=Ninja
[tool_requires]
```

Relevant line (maybe):
```
/bin/sh ../libtool --mode=compile --tag=RC C:/Program Files (x86)/Windows Kits/10/bin/10.0.22000.0/x64/rc.exe `/bin/sh /c/users/joris/.conan2/p/b/libic587608ee0597f/b/src/lib/../windows/windres-options --escape 1.17` -i /c/users/joris/.conan2/p/b/libic587608ee0597f/b/src/lib/../windows/libiconv.rc -o libiconv.res.lo --output-format=coff
/bin/sh: -c: line 1: syntax error near unexpected token `('
/bin/sh: -c: line 1: `/bin/sh ../libtool --mode=compile --tag=RC C:/Program Files (x86)/Windows Kits/10/bin/10.0.22000.0/x64/rc.exe `/bin/sh /c/users/joris/.conan2/p/b/libic587608ee0597f/b/src/lib/../windows/windres-options --escape 1.17` -i /c/users/joris/.conan2/p/b/libic587608ee0597f/b/src/lib/../windows/libiconv.rc -o libiconv.res.lo --output-format=coff'
make[1]: *** [Makefile:99: libiconv.res.lo] Error 2
```

You have windows kit installed in `Program Files (x86)` too?

**Comment by shun2wang** [comment]
Yes I have. but may not related. this libiconv sometime failed and you can comment it out from `conanfile.txt` try build and then add it back try again...

if succeed you can search `iconv` in messages and will see:
```
    libiconv/1.17#fa54397801cd96911a8294bc5fc76335 - Cache
    libiconv/1.17#fa54397801cd96911a8294bc5fc76335:7bfde258ff4f62f75668d0896dbddedaa7480a0f#044260f1a5e41ff6d695c50a8d622149 - Cache
libiconv/1.17: Already installed! (8 of 15)
WARN: deprecated:     'cpp_info.names' used in: libiconv/1.17, zstd/1.5.2, gmp/6.3.0, boost/1.83.0, sqlite3/3.38.5, zlib/1.2.13, libarchive/3.6.2, openssl/3.0.10, brotli/1.0.9, bzip2/1.0.8
WARN: deprecated:     'env_info' used in: m4/1.4.19, libiconv/1.17, boost/1.83.0, sqlite3/3.38.5, bison/3.7.6, openssl/3.0.10, bzip2/1.0.8
-- [Libraries] Conan: Target declared 'Iconv::Iconv'
-- [Libraries] Looking for libiconv-2.dll
-- [Libraries] Looking for libiconv-2.dll - found
-- [Libraries]   C:/rtools43/ucrt64/bin/libiconv-2.dll
```

**Review by JorisGoosen — APPROVED** [review]

**Pull request merged**